py-polars/polars/eager/series.py

```diff
@@ -2089,6 +2089,120 @@ def mode(self) -> "Series":
         """
         return wrap_s(self._s.mode())
 
+    def sin(self) -> "Series":
+        """
+        Compute the element-wise value for Trigonometric sine.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> s = pl.Series("a", np.array((0., np.pi/2., np.pi)))
+        >>> s.sin()
+        shape: (3,)
+        Series: 'a' [f64]
+        [
+                0.0
+                1
+                1.2246467991473532e-16
+        ]
+        """
+        return wrap_s(self._s.apply_lambda(np.sin, Float64))
```

Comment:
Series implement numpy `__ufunc__`, so I believe we can do `return np.sin(self)`. This would be vectorized instead of an elementwise application.